File: my-app/backend/API/Models/OrderAggregate/Order.cs
namespace API.Models.OrderAggregate
{
    public class Order
    {
        public int Id { get; set; }
        public string CustomerId { get; set; }
        public ShippingAddress ShippingAddress { get; set; }
        public DateTime OrderDate { get; set; }
        public List<OrderItem> OrderItems { get; set; }
        public double Subtotal { get; set; }
        public OrderStatus OrderStatus { get; set; } = OrderStatus.Pending;
    }
}

File: my-app/react-app/src/models/order.ts
export interface ShippingAddress {
  fullName: string
  country: string
  detailedAddress: string
  city: string
  province: string
  postalCode: string
}

export interface OrderItem {
  productId: number
  name: string
  imgUrl: string
  price: number
  quantity: number
}

export interface RootObject {
  id: number
  customerId: string
  shippingAddress: ShippingAddress
  orderDate: Date
  orderItems: OrderItem[]
  subtotal: number
  orderStatus: string
  total: number
}
